recipes-browser/wpebackend-rdk/wpebackend-rdk.inc

```diff
@@ -6,16 +6,31 @@ LICENSE = "BSD"
 LIC_FILES_CHKSUM = "file://src/wayland/display.h;;beginline=5;endline=24;md5=b1c8cb6b0857048a21b33611f01c575a"
 
 SRC_URI = "git://github.com/WebPlatformForEmbedded/WPEBackend-rdk.git;protocol=git;branch=master"
+S = "${WORKDIR}/git"
 
 PROVIDES += "virtual/wpebackend"
 RPROVIDES_${PN} += "virtual/wpebackend"
 
-DEPENDS = "libwpe glib-2.0 libinput"
+DEPENDS = "libwpe glib-2.0"
 
 inherit cmake pkgconfig
 
+PACKAGECONFIG ?= "wayland virtualinput"
 
-PACKAGECONFIG ?= "wayland"
+# Wayland selectors
+PACKAGECONFIG[wpeframework] = "-DUSE_BACKEND_WPEFRAMEWORK=ON -DUSE_KEY_INPUT_HANDLING_LINUX_INPUT=OFF -DUSE_INPUT_LIBINPUT=OFF,,wpeframework-clientlibraries wpeframework-plugins libxkbcommon xkeyboard-config"
+
+# other switches
+PACKAGECONFIG[debug]            = "-DCMAKE_BUILD_TYPE=Debug,-DCMAKE_BUILD_TYPE=Release,"
+PACKAGECONFIG[libinput]         = "-DUSE_INPUT_LIBINPUT=ON,-DUSE_INPUT_LIBINPUT=OFF,libinput,"
+PACKAGECONFIG[libinput-udev]    = "-DUSE_INPUT_UDEV=ON,-DUSE_INPUT_UDEV=OFF,libinput eudev,"
```

Comment:
Certain backends _require_ usage of libinput to be able to receive input events. For example the `src/viv-imx6/view-backend.cpp` file uses the `LibinputServer` unconditionally, so passing `-DUSE_INPUT_LIBINPUT=OFF` will make the build fail. It looks to me that this kind of cases should be taken into account here.

As for `-DUSE_INPUT_UDEV=` option, what I would do is always enable it if `libudev` is being built. The amount of code added to the backend is very small, and libput will fallback to enumerate itself devices from `/dev/input` if udev is not available at runtime, so there is hardly a reason to have a build option for this. That being said, I have no idea how complicated to do the “enable option X if package Y is being built” with Bitbake, so if it's hard feel free to ignore the suggestion.